lib/chef/provider/machine.rb

```diff
@@ -146,6 +146,12 @@ def machine_options(driver)
           result
         end
     }
+    
+    # I'm making the assumption here that you won't be using 2 different drivers
+    # in the same Ruby instance
```

Comment:
Yeah, I completely agree.  Here is my reasoning:

To add a new attribute I _must_ do it at the class level (unless someone knows differently).  Or I need to introduce a completely new resource and make the `machine` DSL use my new resource rather than the original resource.  Adapting the `resource_mapping` logic to use the `current_driver` to lookup the correct resource is a _lot_ of work for a system that I don't want to maintain.  Eventually I want users to type `aws_instance` instead of `machine`.

So thats why I decided to add `aws_tags` as an attribute to _all_ `machine`s if the AWS driver is loaded.  I also don't know of any users managing multi-provider cookbooks.

Is this explanation reason enough to keep the code as it is?  Would you like me to update the comment to be more informative?